### PR TITLE
ERM inventory count #281 and ERM inventory cost #280 query

### LINKED ISSUE #280: Create ERM-Inventory Costs Cluster query

### LINKED ISSUE #281: Create ERM-Inventory Count Cluster query

### PR DESCRIPTION
**Pull request by adoerrer**
Fixes #281 and fixes #280

### CODE REVIEW

**Comment by nassibnassar** [comment]
@adoerrer Did you want to merge this into the erm branch rather than main?

**Comment by adoerrer** [comment]
> @adoerrer Did you want to merge this into the erm branch rather than main?

No, this can be in the regular main branch. The both new queries are made to count and calculate costs on Electronic Resources that are held in the inventory (#280 & #281) This is a light start into ERM reports.

I also removed the derived tables #116 to #120 which rely on the agreement tables that are not available yet.

File: sql/report_queries/erm/costs/erm_inventory_cost.sql [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        LEFT JOIN invoice_detail AS inv ON invl.invoice_id = inv.invoice_id
+        LEFT JOIN folio_reporting.invoice_adjustments_ext AS inv_adj ON invl.invoice_id = inv_adj.invoice_id
+        LEFT JOIN po_lines AS pol ON invl.po_line_id = pol.id
+        LEFT JOIN instance_detail AS inst ON pol.instance_id = inst.instance_id
+        LEFT JOIN folio_reporting.po_lines_phys_mat_type AS pol_phys_type ON pol.instance_id = pol_phys_type.pol_id
+        LEFT JOIN folio_reporting.po_lines_er_mat_type AS pol_er_type ON pol.instance_id = pol_er_type.pol_id
+        LEFT JOIN folio_reporting.invoice_lines_adjustments AS invl_adjustments ON invl.id = invl_adjustments.invoice_line_id
+        LEFT JOIN instance_subs AS instsub ON instsub.instance_id = inst.instance_id
+        LEFT JOIN instance_formats AS instform ON instform.instance_id = inst.instance_id
+WHERE
```
Comment (jenmawe) [inline sql/report_queries/erm/costs/erm_inventory_cost.sql:130]: I got an error here when I ran the query in DBeaver. I think the problem is further down with one of the "AND"s where there's an extra one.

File: sql/report_queries/erm/costs/erm_inventory_cost.sql [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+             folio_reporting.instance_formats AS forms
+         GROUP BY
+             forms.instance_id
+     ),
+     invoice_detail AS (
+         SELECT
+             inv.id AS "invoice_id",
+             inv.approval_date AS "invoice_approval_date"
+         FROM
+             invoice_invoices AS inv
```
Comment (jenmawe) [inline sql/report_queries/erm/costs/erm_inventory_cost.sql:84]: I ran this against snapshot and snapshot20210114_invoice hoping that there would be some data. In each case, there is no data. There is the empty column id but no data for approval_date. I commented out approval_date which removed the error but not that there is no data.

Comment (adoerrer) [reply sql/report_queries/erm/costs/erm_inventory_cost.sql:84]: Thank you for taking this PR to review and sorry for lacking some informations on how to set a proper test envrionment.
A good choice of testing this query is ` folio_snapshot_20210114_invoice `
But as it is using a new derived table you have to rename the joined `table folio_reporting.invoice_adjustments_ext` to `local.axeldoerrer_invoice_adjustments_ext` .
You then will receive only one row, which is unfortunately the only row which relies on electronic resources. But if you remove the preset filter value for format_name you'll see more rows which should give you an idea how that report is intended.

**Review by jenmawe — APPROVED** [review]
Axel showed me how this works.

File: sql/report_queries/erm/count/erm_inventory_count.sql [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    folio_reporting.instance_ext AS inst
+        LEFT JOIN folio_reporting.instance_statistical_codes AS insc ON inst.instance_id = insc.instance_id
+        LEFT JOIN folio_reporting.instance_nature_content AS innc ON inst.instance_id = innc.instance_id
+        LEFT JOIN folio_reporting.instance_formats AS inform ON inst.instance_id = inform.instance_id
+        LEFT JOIN folio_reporting.instance_relationships_ext AS super_relation ON super_relation.relationship_super_instance_id = inst.instance_id
+        LEFT JOIN folio_reporting.instance_relationships_ext AS sub_relation ON sub_relation.relationship_sub_instance_id = inst.instance_id
+        LEFT JOIN folio_reporting.holdings_ext AS hld ON inst.instance_id = hld.instance_id
+        LEFT JOIN folio_reporting.holdings_statistical_codes AS hsc ON hld.holdings_id = hsc.holdings_id
+        LEFT JOIN folio_reporting.locations_libraries AS loc ON hld.permanent_location_id = loc.location_id
+        LEFT JOIN folio_reporting.instance_languages AS lng ON lng.instance_id = inst.instance_id AND lng.ordinality = 1
```
Comment (nassibnassar) [inline sql/report_queries/erm/count/erm_inventory_count.sql:103]: `lng.language_ordinality`

File: sql/report_queries/erm/count/erm_inventory_count.sql [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+            (NOT inst.discovery_suppress)
+            OR (inst.discovery_suppress ISNULL))
+  AND
+  -- filter all virtual titles (update values as needed).
+    (inform.format_name IN  ('computer -- online resource')
+        OR  loc.library_name  IN  ('Online'))
+  AND
+
+  -- begin to process the set filters
+  AND
```
Comment (nassibnassar) [inline sql/report_queries/erm/count/erm_inventory_count.sql:119]: Syntax error

**Review by nassibnassar — CHANGES_REQUESTED** [review]

File: sql/report_queries/erm/count/erm_inventory_count.sql (hunk 3 above)
Comment (adoerrer) [reply sql/report_queries/erm/count/erm_inventory_count.sql]: ok

File: sql/report_queries/erm/count/erm_inventory_count.sql (hunk 4 above)
Comment (adoerrer) [reply sql/report_queries/erm/count/erm_inventory_count.sql]: ok

**Review by nassibnassar — APPROVED** [review]

**Pull request merged**